### PR TITLE
fix: Normalize Firefly url

### PR DESCRIPTION
**Pull request by luispresuelVenafi**

### CODE REVIEW

File: pkg/venafi/firefly/connector.go [hunk 1]
```diff
@@ -59,7 +60,18 @@ func (c *Connector) RetrieveAvailableSSHTemplates() (response []certificate.SshA
...
 func NewConnector(url string, zone string, verbose bool, trust *x509.CertPool) (*Connector, error) {
-	return &Connector{baseURL: url, zone: zone, verbose: verbose, trust: trust}, nil
+	normalizedUrl, err := normalizeURL(url)
+	if err != nil {
+		return nil, fmt.Errorf("%w: failed to normalize URL: %v", verror.UserDataError, err)
+	}
+	return &Connector{baseURL: normalizedUrl, zone: zone, verbose: verbose, trust: trust}, nil
+}
+
+func normalizeURL(url string) (normalizedURL string, err error) {
```
Comment (EduardoVV) [inline pkg/venafi/firefly/connector.go:70]: Is this extra function needed?

Comment (luispresuelVenafi) [reply pkg/venafi/firefly/connector.go:70]: Good catch

Comment (luispresuelVenafi) [reply pkg/venafi/firefly/connector.go]: @EduardoVV fixed by latest commit

**Review by EduardoVV — APPROVED** [review]

File: pkg/venafi/cloud/connector.go [hunk 2]
```diff
@@ -673,16 +672,9 @@ func normalizeURL(url string) (normalizedURL string, err error) {
 		url = apiURL
 		//return "", fmt.Errorf("base URL cannot be empty")
```
Comment (rvelaVenafi) [inline pkg/venafi/cloud/connector.go:673]: Remove this comment as no longer needed or return right away as no further modifications are required to the default URL

File: pkg/venafi/cloud/connector.go [hunk 3]
```diff
@@ -673,16 +672,9 @@ func normalizeURL(url string) (normalizedURL string, err error) {
...
-	} else {
-		modified = reg.ReplaceAllString(modified, "https://")
-	}
-	if !strings.HasSuffix(modified, "/") {
-		modified = modified + "/"
-	}
+
+	modified := util.NormalizeUrl(url)
+
 	normalizedURL = modified
```
Comment (rvelaVenafi) [inline pkg/venafi/cloud/connector.go:678]: This step can be removed, there is no further modification to the url after returning from `util.NormalizeUrl()` function

File: pkg/venafi/firefly/connector.go [hunk 4]
```diff
@@ -25,6 +25,7 @@ import (
 	"strings"
 	"time"
 
+	"github.com/Venafi/vcert/v5/pkg/util"
```
Comment (rvelaVenafi) [inline pkg/venafi/firefly/connector.go:28]: Sort this import accordingly to our rules

File: pkg/venafi/firefly/connector.go [hunk 5]
```diff
@@ -59,7 +60,8 @@ func (c *Connector) RetrieveAvailableSSHTemplates() (response []certificate.SshA
 
 // NewConnector creates a new Firefly Connector object used to communicate with Firefly
 func NewConnector(url string, zone string, verbose bool, trust *x509.CertPool) (*Connector, error) {
-	return &Connector{baseURL: url, zone: zone, verbose: verbose, trust: trust}, nil
+	normalizedUrl := util.NormalizeUrl(url)
```
Comment (rvelaVenafi) [inline pkg/venafi/firefly/connector.go:63]: Move this line to a NormalizeUrl function like in other connectors and add a validation for empty url. Just like other connectors, make the NormalizeUrl return an error and use it here in the constructor.

**Comment by rvelaVenafi** [comment]
Also, I noticed NewConnector() in cloud connector just returns the error when normalizeUrl fails. While TPP wraps the error. We should standardize and use the same error as TPP is using for Cloud and Firefly connectors in their constructors.

**Review by rvelaVenafi — CHANGES_REQUESTED** [review]

File: pkg/venafi/firefly/connector.go (hunk 4 above)
Comment (marcos-albornoz) [reply pkg/venafi/firefly/connector.go]: It's done

File: pkg/venafi/cloud/connector.go (hunk 3 above)
Comment (marcos-albornoz) [reply pkg/venafi/cloud/connector.go]: It's done

File: pkg/venafi/cloud/connector.go (hunk 2 above)
Comment (marcos-albornoz) [reply pkg/venafi/cloud/connector.go]: It's done

File: pkg/venafi/firefly/connector.go (hunk 5 above)
Comment (marcos-albornoz) [reply pkg/venafi/firefly/connector.go]: It's done

File: pkg/venafi/firefly/connector.go [hunk 6]
```diff
@@ -59,9 +60,22 @@ func (c *Connector) RetrieveAvailableSSHTemplates() (response []certificate.SshA
 
 // NewConnector creates a new Firefly Connector object used to communicate with Firefly
 func NewConnector(url string, zone string, verbose bool, trust *x509.CertPool) (*Connector, error) {
+	if url != "" {
+		var err error
```
Comment (rvelaVenafi) [inline pkg/venafi/firefly/connector.go:64]: This declaration is not required.
below line can do
`url, err := normalizeUrl()`

**Review by rvelaVenafi — CHANGES_REQUESTED** [review]

File: pkg/venafi/firefly/connector.go (hunk 6 above)
Comment (marcos-albornoz) [reply pkg/venafi/firefly/connector.go:64]: that is not correct, given that create a new `user` var which only lives in the scope of the surrounder `if condition`

Comment (rvelaVenafi) [reply pkg/venafi/firefly/connector.go:64]: got it

**Review by rvelaVenafi — APPROVED** [review]

**Pull request merged**